build/Targets/Settings.props

```diff
@@ -81,6 +81,16 @@
     <ProjectAssetsFile Condition="'$(BuildingForLiveUnitTesting)' == 'true'">$(LiveUnitTestingOriginalBaseIntermediateOutputPath)project.assets.json</ProjectAssetsFile>
   </PropertyGroup>
 
+  <!-- Utilize source linking whenever there is a valid git commit available. Won't be available during rebase, 
+       or build from zip for example -->
+  <PropertyGroup Condition="'$(GitHeadShaRaw)' != '' AND '$(OS)' == 'Windows_NT' ">
+    <SourceLinkCreate>true</SourceLinkCreate>
+    <SourceLinkRootDirectory Condition="'$(SourceLinkRootDirectory)' == ''">$(RepoRoot)</SourceLinkRootDirectory>
+    <SourceLinkOriginUrl>https://github.com/dotnet/roslyn</SourceLinkOriginUrl>
+    <SourceLinkUrl>https://raw.githubusercontent.com/dotnet/roslyn/$(GitHeadShaRaw)/*</SourceLinkUrl>
+    <SourceLinkCommit>$(GitHeadShaRaw)</SourceLinkCommit>
```

Comment:
If `SourceLinkUrl` and the `SourceLinkCommit` are not set, they are figured out by running git commands. Setting everything works too. Reviewing the [targets](https://github.com/ctaggart/SourceLink/blob/v2/SourceLink.Create.CommandLine/SourceLink.Create.CommandLine.targets), you could make some minor changes.

- SourceLinkOriginUrl isn't needed because you are setting SourceLinkUrl
- SourceLinkUrl [will replace](https://github.com/ctaggart/SourceLink/blob/v2/SourceLink.Create.CommandLine/CreateTask.cs#L56) `{commit}` with the value from `SourceLinkCommit`.

``` xml
  <PropertyGroup Condition="'$(GitHeadShaRaw)' != '' AND '$(OS)' == 'Windows_NT' ">
    <SourceLinkCreate>true</SourceLinkCreate>
    <SourceLinkRootDirectory Condition="'$(SourceLinkRootDirectory)' == ''">$(RepoRoot)</SourceLinkRootDirectory>
    <SourceLinkUrl>https://raw.githubusercontent.com/dotnet/roslyn/{commit}/*</SourceLinkUrl>
    <SourceLinkCommit>$(GitHeadShaRaw)</SourceLinkCommit>
  </PropertyGroup>
```

Looks great otherwise!